sdk/formrecognizer/azure-ai-formrecognizer/src/main/java/com/azure/ai/formrecognizer/implementation/Utility.java

```diff
@@ -16,37 +17,107 @@
  */
 public final class Utility {
     private static final ClientLogger LOGGER = new ClientLogger(Utility.class);
+    // using 4K as default buffer size: https://stackoverflow.com/a/237495/1473510
+    private static final int BYTE_BUFFER_CHUNK_SIZE = 4096;
 
     private Utility() {
     }
 
     /**
-     * A utility method for converting the input stream to Flux of ByteBuffer.
+     * Automatically detect byte buffer's content type.
+     * Given the source: <a href="https://en.wikipedia.org/wiki/Magic_number_(programming)#Magic_numbers_in_files"/>.
      *
-     * @param data The input data which needs to convert to ByteBuffer.
+     * @param buffer The byte buffer input.
      *
-     * @return {@link ByteBuffer} which contains the input data.
-     * @throws RuntimeException When I/O error occurs.
+     * @return The {@link ContentType} content type.
      */
-    public static Flux<ByteBuffer> convertStreamToByteBuffer(InputStream data) {
-        return Flux.just(toByteArray(data))
-            .doOnError(error -> LOGGER.warning("Failed to convert stream to byte array - {}", error));
+    public static ContentType getContentType(ByteBuffer buffer) {
+        final byte[] bytes = buffer.array();
+        if (bytes.length < 4) {
```

Comment:
Since we are only checking the first `ByteBuffer` what would happen if the first buffer is empty due to a disk or networking latency but then the next one contains the byte marking which would properly determine the file type.